plugins/git4idea/src/git4idea/branch/GitBranchUtil.java

```diff
@@ -75,8 +75,9 @@ private GitBranchUtil() {}
    */
   @Nullable
   public static GitBranchTrackInfo getTrackInfoForBranch(@NotNull GitRepository repository, @NotNull GitLocalBranch branch) {
+    GitRemoteBranch remoteBranch = branch.findTrackedBranch(repository);
```

Comment:
This looks a bit weird, since `branch.findTrackedBranch` walks through `getBranchTrackInfos` inside.

Btw, I've roughly checked the usages, and it seems that all of them will be OK to receive a `GitRemoteBranch` object (which luckily has `getRemote()`) instead of the whole `GitTrackInfo`.

I propose to use `branch.findTrackedBranch` in the push dialog and other usages. I propose to check which usages are affected by symrefs and fix them, leaving other as is, and I'll fix them later.